test-float-features.lisp

```diff
@@ -0,0 +1,107 @@
+(in-package #:cl-user)
+
+(defpackage #:float-features/tests
+  (:nicknames #:org.shirakumo.float-features/tests)
+  (:use #:cl))
+
+;;; to run (parachute:test :float-features/tests)
+(in-package #:float-features/tests)
+
+(parachute:define-test float-features-1
+    :compile-at :execute
+    (parachute:true
+     (float-features:float-infinity-p
+      (flet ((foo () (if (> 10 (random 20)) 0.0 0.0))
+             (bar () (if (> 10 (random 20)) 23 24)))
```

Comment:
It is not a good idea to introduce randomness into tests like this. They are not executed multiple times in a suite run, making the test indeterministic.

The same goes for all the other tests in this suite.

Reply:
This was to avoid the clever constant folding of sbcl, but I believe this can be tackled with the following
`Sometimes the compiler will already complain for some tests that you expect to fail, for instance when the type inference is too good. In that case you can force Parachute to only compile the test forms when the test is evaluated. `

Reply:
An deterministc way to circumvent constant folding would be to use something like `read-from-string`

Reply:
done